### PR TITLE
Refactor company descriptions

### PR DESCRIPTION
**Pull request by Jack-Vines**
This PR adds a section on how to train a company description classifier.

Don't worry too much about actually running the flows as they're copied from a different location and have already been reviewed, but have a look through the README to check it all makes sense!

### CODE REVIEW

**Comment by cmbrennan002** [comment]
I had a read through, looked very clear to me! As agreed on slack, I've added the content-related questions as an issue for future updates, and otherwise I didn't have any other suggestions :)

File: company_descriptions/make_training_data/split_job_ads_flow.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        """Cleans job ads by:
+        - removing numbers and stop words
+        - Creating new sentences by Camel Case
+        - splitting sentences
+
+        and saves sentence chunks as .csv files
+        """
+        import uuid
+
+        from nesta_ds_utils.loading_saving.S3 import upload_obj
```
Comment (RFOxbury) [inline company_descriptions/make_training_data/split_job_ads_flow.py:75]: Just for my understanding - what's the benefit of importing these libraries within a step, rather than at the top of the script?

File: company_descriptions/compdescs_flow.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+        for config_name, config_value in CONFIG.items():
+            self.trainer_results[config_name] = config_value
+
+        # convert logits to probabilities
+        probs = softmax(trainer.predict(self.tokenized_ds["test"]).predictions, axis=1)
+        # convert probabilities to predicted labels
+        y_pred = np.argmax(probs, axis=-1)
+        # get true labels
+        y_true = self.tokenized_ds["test"]["label"]
```
Comment (RFOxbury) [inline company_descriptions/compdescs_flow.py:179]: I might have missed something but should the test set be separate from the dataset used as `eval_dataset` at l.161? It looks like the same dataset is being used as part of evaluation during the fine-tuning process, and also to generate the final metrics. But I think there should be a separate hold-out test set, right?

**Review by RFOxbury — COMMENTED** [review]
I left one question about train/validation/test sets but I think maybe I just haven't understood something. But then if that's the case, it might be good to have some extra comments in the code just so that everything is super clear!
